### PR TITLE
Pbi 642

### PR DESCRIPTION
**Pull request by aeng7**
-initial batch of IMA unit tests
-please check out pbi-642b if you want to try the tests out. The other branches have poor diff readability because they were branched off of another branch. pbi-642b is branch that was rebased off of master.

### CODE REVIEW

**Comment by aeng7** [comment]
Please also check to see if these unit tests pass on your machines. If something fails, please include the node version you have and the tests that fail. Thanks!

File: js/google_ima.js [hunk 1]
```diff
@@ -1216,7 +1219,7 @@ require("../html5-common/js/utils/utils.js");
       /**
        * Callback from IMA SDK for ad tracking events.
        * @private
-       * @method GoogleIMA#_IMA_SDK_resumeMainContent
+       * @method GoogleIMA#_IMA_SDK_onAdEvent
```
Comment (pilievOoyala) [inline js/google_ima.js]: Thank you for fixing that!

File: js/google_ima.js [hunk 2]
```diff
@@ -1329,17 +1332,21 @@ require("../html5-common/js/utils/utils.js");
       var _startTimeUpdater = privateMember(function()
       {
         _stopTimeUpdater();
-        _timeUpdater = setInterval(_.bind(function()
+        //starting an interval causes unit tests to throw a max call stack exceeded error
```
Comment (pilievOoyala) [inline js/google_ima.js]: Add note that we don't need to test this with unit tests yet. and add the date.

**Comment by pilievOoyala** [comment]
The branch to check out is pbi-642b

File: test/unit-tests/ima_test.js [hunk 3]
```diff
@@ -119,20 +187,703 @@ describe('ad_manager_ima', function() {
...
+    };
+    ima.playAd({
+      ad : {}
+    });
+    var am = google.ima.adManagerInstance;
+    am.publishEvent(google.ima.AdEvent.Type.STARTED);
+    am.publishEvent(google.ima.AdEvent.Type.AD_ERROR);
+    expect(linearAdEndedNotified).to.be(true);
+    expect(podEndedNotified).to.be(true);
+    expect(doneControllingAdsNotified).to.be(true);
```
Comment (pilievOoyala) [inline test/unit-tests/ima_test.js]: I think we need to clean up functions that you add to the amc for a specific test. Or does the amc get recreated for each test?

**Comment by pilievOoyala** [comment]
I'm going to be that guy.  Since we are trying to start the new coding style, can you put braces '{' on new lines?

File: test/unit-tests/ima_test.js (hunk 3 above)
Comment (aeng7) [inline test/unit-tests/ima_test.js]: Amc gets recreated for each test in the afterEach function with the amc = originalMockAmc; line.

**Comment by aeng7** [comment]
Will do on the braces.

**Comment by pilievOoyala** [comment]
LGTM

**Pull request merged**